src/bbs-plus/params.ts

```diff
@@ -133,42 +143,127 @@ export class SignatureParamsG1 extends SignatureParams {
     blinding?: Uint8Array
   ): [Uint8Array, Uint8Array] {
     const b = blinding === undefined ? generateRandomFieldElement() : blinding;
-    const commitment = bbsCommitMsgsInG1(messageToCommit, b, this.value, encodeMessages);
+    const commitment = bbsPlusCommitMsgsInG1(messageToCommit, b, this.value, encodeMessages);
     return [commitment, b];
   }
+
+  static signMessageObject(
+    messages: Object,
+    secretKey: BBSPlusSecretKey,
+    labelOrParams: Uint8Array | BBSPlusSignatureParamsG1,
+    encoder: Encoder
+  ): SignedMessages<BBSPlusSignatureG1> {
+    const [names, encodedValues] = encoder.encodeMessageObject(messages);
+    const msgCount = names.length;
+  
+    const sigParams = this.getSigParamsOfRequiredSize(msgCount, labelOrParams);
+    const signature = BBSPlusSignatureG1.generate(encodedValues, secretKey, sigParams, false);
+  
+    // Encoded message as an object with key as the flattened name
+    const encodedMessages: { [key: string]: Uint8Array } = {};
+    for (let i = 0; i < msgCount; i++) {
+      encodedMessages[names[i]] = encodedValues[i];
+    }
+  
+    return {
+      encodedMessages,
+      signature
+    };
+  }
+
+  /**
+   * Verifies the signature on the given messages. Takes the messages as a JS object, flattens it, encodes the values similar
+   * to signing and then verifies the signature.
+   * @param messages
+   * @param signature
+   * @param publicKey
+   * @param labelOrParams
+   * @param encoder
+   */
+  static verifyMessageObject(
+    messages: object,
+    signature: BBSPlusSignatureG1,
+    publicKey: BBSPlusPublicKeyG1,
+    labelOrParams: Uint8Array | BBSPlusSignatureParamsG1,
+    encoder: Encoder
+  ): VerifyResult {
+    const [_, encodedValues] = encoder.encodeMessageObject(messages);
+    const msgCount = encodedValues.length;
+
+    const sigParams = this.getSigParamsOfRequiredSize(msgCount, labelOrParams);
+    return signature.verify(encodedValues, publicKey, sigParams, false);
+  }
+
+  /**
+  * Gives `SignatureParamsG1` that can sign `msgCount` number of messages.
+  * @param msgCount
+  * @param labelOrParams
+  */
+  static getSigParamsOfRequiredSize(
```

Comment:
The problem is with `labelOrParams: Uint8Array | BBSSignatureParams`, so we should know what type must be constructed in case a label is provided.
But I think about recreating `ISignatureParams` as an abstract class and moving there all repeated methods implementation.